core/src/mender-api.c

```diff
@@ -337,10 +346,7 @@ api_check_for_deployment_v1(int *status, void *response) {
     char        *path = NULL;
 
     /* Compute path */
-    /* TODO: Retrieve artifact name from store (see ticket MEN-7479) */
-    if (-1
-        == asprintf(
-            &path, MENDER_API_PATH_GET_NEXT_DEPLOYMENT "?artifact_name=%s&device_type=%s", mender_api_config.artifact_name, mender_api_config.device_type)) {
+    if (-1 == asprintf(&path, MENDER_API_PATH_GET_NEXT_DEPLOYMENT "?artifact_name=%s&device_type=%s", artifact_name, mender_api_config.device_type)) {
```

Comment:
I think we should add `assert(NULL != artifact_name)` to the beginning of all functions that assume so.